kmod/core/core.c

```diff
@@ -661,31 +699,26 @@ static int kpatch_write_relocations(struct kpatch_module *kpmod,
        unsigned long core = (unsigned long)kpmod->mod->module_core;
        unsigned long core_size = kpmod->mod->core_size;
 #endif
-	unsigned long src;
+	bool vmlinux = !strcmp(object->name, "vmlinux");
 
 	list_for_each_entry(dynrela, &object->dynrelas, list) {
-		if (!strcmp(object->name, "vmlinux")) {
-			ret = kpatch_verify_symbol_match(dynrela->name,
-							 dynrela->src);
-			if (ret)
-				return ret;
+		if (vmlinux || !dynrela->external) {
```

Comment:
@flaming-toast : In v2 I left the conditional ordering the same, but you're right, this arrangement would be make more readable source.